hrm-domain/hrm-service/migrations/20250721194303-add_cases_and_profiles_definitionVersion.js

```diff
@@ -0,0 +1,127 @@
+        SELECT DISTINCT ON (c."accountSid")
+          c."accountSid",
+          c."definitionVersion"
+        FROM "Contacts" c
+        ORDER BY c."accountSid", c."createdAt" DESC
+      )
+
+      UPDATE "Cases"
+      SET "definitionVersion" = COALESCE(c."info"->>'definitionVersion', COALESCE(ocpc."definitionVersion", lcpa."definitionVersion"))
+      FROM "Cases" c
+      LEFT JOIN oldest_contact_per_case ocpc ON c.id = ocpc."caseId"
+      LEFT JOIN latest_contact_per_account lcpa ON c."accountSid" = lcpa."accountSid"
+      WHERE "Cases".id = c.id;
+    `);
+    console.log('"definitionVersion" column populated in table "Cases"');
+
+    await queryInterface.sequelize.query(`
+      UPDATE "Cases" SET "definitionVersion" = 'as-v1' WHERE "definitionVersion" = 'demo-v1';
+      UPDATE "Cases" SET "definitionVersion" = 'zm-v1' WHERE "definitionVersion" = 'v1';
+    `);
+    console.log('"definitionVersion" column sanitized in table "Cases"');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE "Profiles" ADD COLUMN IF NOT EXISTS "definitionVersion" TEXT;
+    `);
+    console.log('"definitionVersion" column added to table "Profiles"');
+
+    await queryInterface.sequelize.query(`
+      WITH
+      -- Oldest contact per profile
+      oldest_contact_per_profile AS (
+        SELECT DISTINCT ON (c."profileId")
+          c."profileId",
+          c."definitionVersion"
+        FROM "Contacts" c
+        ORDER BY c."profileId", c."createdAt" ASC
+      ),
+
+      -- Latest contact per account (for orphan fallback)
+      latest_contact_per_account AS (
+        SELECT DISTINCT ON (c."accountSid")
+          c."accountSid",
+          c."definitionVersion"
+        FROM "Contacts" c
+        ORDER BY c."accountSid", c."createdAt" DESC
+      )
+
+      UPDATE "Profiles"
+      SET "definitionVersion" = COALESCE(ocpp."definitionVersion", lcpa."definitionVersion")
+      FROM "Profiles" p
+      LEFT JOIN oldest_contact_per_profile ocpp ON p.id = ocpp."profileId"
+      LEFT JOIN latest_contact_per_account lcpa ON p."accountSid" = lcpa."accountSid"
+      WHERE "Profiles".id = p.id;
+    `);
+    console.log('"definitionVersion" column populated in table "Profiles"');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE public."Contacts" ALTER COLUMN "definitionVersion" SET NOT NULL;    
+    `);
+    console.log('"definitionVersion" column for "Contacts" table set to not null');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE public."Cases" ALTER COLUMN "definitionVersion" SET NOT NULL;    
+    `);
+    console.log('"definitionVersion" column for "Cases" table set to not null');
+
+    await queryInterface.sequelize.query(`
+      ALTER TABLE public."Profiles" ALTER COLUMN "definitionVersion" SET NOT NULL;    
+    `);
+    console.log('"definitionVersion" column for "Profiles" table set to not null');
+  },
+  down: async queryInterface => {
+    await queryInterface.sequelize.query(`
```

Comment:
Do we need to set the definitionVersion column on Contacts back to being nullable again?